freqtrade/persistence/models.py

```diff
@@ -259,16 +264,31 @@ class LocalTrade():
     strategy: str = ''
     timeframe: Optional[int] = None
 
-    def __init__(self, **kwargs):
-        for key in kwargs:
-            setattr(self, key, kwargs[key])
-        self.recalc_open_trade_value()
+    # Margin trading properties
+    interest_rate: float = 0.0
+    liquidation_price: Optional[float] = None
```

Comment:
In isolated trading mode it won't change. I'm not exacting sure how it works for cross trading mode or on exchanges other than binance

Reply:
I don't think we'll be supporting isolated trading mode at all.

it'll require to transfer funds to that "pair's margin wallet" - and as such will be impossible to handle when trading multiple pairs.
assume i trade 30 pairs - i'd have to have balance in each of these isolated pairs - which will make utilizing the whole balance almost impossible.

Reply:
What makes it impossible? There would just be a few actions to execute with each trade, a transfer, a borrow, a trade, then a trade, a repayment, and a transfer. I was planning on implementing this because the leverage is a lot higher, I still don't really understand futures, but I guess I should look into that

Reply:
To be honest, i'm tempted to start with 0 leverage (only allowing short) for the beginning - although i'm not sure that's possible on binance (short without leverage).

Adding isolated margin support might be a future enhancement, but is nothing i'll be accepting as part of the PR adding this functionality to start with.

same goes for futures i'd say (at least for non-perpetual futures) - as they will add a whole other level of complexity (expiration dates).

Reply:
There's always something borrowed when you're shorting on any exchange, you're selling something you don't have, with the hopes of buying it again when it's cheaper so there will always be a liquidation price for shorts

Reply:
yeah that's correct.

however liquidation price will change independently of the current trade - so are we sure that storing that as part of the trade object is a good idea?

Reply:
So I think cross margin is actually the more complicated one. 

With cross margin, you need to keep track of the overall equity in your wallet/account, and if your equity drops below a certain threshold, then all trades are liquidated. Is freqtrade able to keep track of prices in real time like that to know if the account equity has dropped passed that point? If it can't then we need to ask the exchange or get some kind of notification from it if liquidation is starting to happen. If the liquidation price is passed we need to exit all trades.

With isolated margin, there's just one ccxt command to transfer money in, and one to transfer out on buy and sell, and then the liquidation price is static unless money is part of the trade is sold/short_exited

`response = binance.transfer('USDT', 16, 'main', 'margin')`

---------------

Isolated perpetual futures seems like it might actually be the least complicated, because it doesn't involve any transferring of funds. You put the money is futures, and can trade that money on any isolated pair. I still don't completely understand futures though.

-------------

Here's an[ explanation of how the liquidation price is calculated on cross margin with binance](https://www.binance.com/en/support/faq/f6b010588e55413aa58b7d63ee0125ed).


I don't think that starting with cross-margin is a way to make implementing this less complicated is the main point I'm trying to make

Reply:
So if isolated trading is used, then `liquidation_price` should be on the `Trade`/`LocalTrade` class (maybe the name should be `isolated_liq` ?), if cross margin is used, then we will have to create a `maintenance_margin` variable(which really sounds like it could be it's own class) inside of `FreqtradeBot` or something, but cross margin isn't something I think I would personally use and I would rather leave it out of these PRs